### DISCUSSION TITLE
Redlining-Tool zum Kreis zeichnen

### DISCUSSION THREAD

**Discussion by k-placer** [e-netze/webgis-community#41] [Ideas]
In webgis4 konnte man mit dem Redlining-Werkzeug "Punkt" ein **Punkt** in einer definierten **Farbe** und **Größe** einzuzeichnen:
<img width="176" alt="redlining_kreis_zeichnen" src="https://user-images.githubusercontent.com/97947925/220857845-ba2ace94-2a01-4b6f-b660-eab5c0b7866d.PNG">

Eine (oder mehrere) der folgenden Möglichkeiten wären als Ersatz denkbar, sind aber momentan in der Benutzeroberfläche nicht verfügbar:
- Bei "Kreis" die Option Radius (z.B. Radius = 5m) durch Eintippen einer Zahl selber einstellen zu können (Bevorzugte Variante!)
- Bei "Symbol" die zusätzliche Option zum Größe ändern
- Bei "Umgebungskreis" die zusätzliche Option zum "Beschriftung deaktivieren"

**Comment by gis-eni** [comment]
Ja, die Punkt sind rausgeflogen, weil es diese als Grafik im Leaflet nicht wirklich gibt. Ein Punkt ist hier immer ein Marker. Das verwenden wir bei den Symbolen. Wenn du also wirklich Punkte haben willst, sollten wie diese bei den Marken hinzufügen. Die Symbol, die du zur Auswahl bekommst liegen unter ``api/wwwroot/content/api/img/graphics/markers/*.(png|gif)``

Hier kannst du dir einfach einmal Punkt hinzufügen. 

Wenn du wirklich "Kreise" machen willst, kannst du das jetzt schon:
1. Mittelpunkt zeichen
2. Rechte Maustaste => Stützpunkt konstruieren => Richtung/Entfernen
3. Bei Entfernung 5 eingeben

Beim nächsten Punkt ist dieser Radius dann der Standard: 
1. Mittelpunkt setzen
2. Direkt auf "Übernehmen/Weiteres zeichnen" Klicken

**Reply by k-placer** [reply]
Das Kontext Menü ist für fortgeschrittene User*innen. Gerade hier ist es besonders wichtig, dass die Beschriftungstexte verständlich formuliert und gut lesbar sind. 

Der Beschriftungstext "Richtung/Entfernung" ist leider abgeschnitten. "Entfernung" fehlt. Kann der Beschriftungstext so angepasst werden, dass beide Texte ("Richtung", "Entfernung") gleichzeitig zu lesen sind?
<img width="308" alt="Kontext_Menue_Richtung_Entfernung" src="https://user-images.githubusercontent.com/97947925/220915594-90e0b2c6-9c06-446d-807a-a665c5375e72.PNG">

**Comment by gis-eni** [comment]
Wenns einfach bleiben sollte, dann ist vielleicht die Variante mit den vorgefertigten Symbolen die bessere?

**Reply by k-placer** [reply]
Für eine einfache Lösung kann ich mir die Variante mit den vorgefertigten Symbolen durchaus vorstellen. 

Davon einmal abgesehen:
Die abgeschnittene Beschriftung bei Tools sieht nicht professionell aus und ist nicht selbsterklärend. Kann man diese abgeschnittenen Beschriftungstexte weglassen? Beim MouseOver sehe ich den vollständigen Beschreibungstext sowieso.

**Comment by gis-eni** [comment]
Du kannst deine Stylesheets erweitern und diese Texte einfache auf unsichtbar setzen: und vielleicht zusätzlich auch die Höhe der Buttons anpassen.

``` css
.webgis-ui-imagebutton .webgis-ui-imagebutton-text {
    display: none;
}
.webgis-ui-optionscontainer.contains-lables .webgis-ui-imagebutton {
    height: 40px;
}

```

![image](https://user-images.githubusercontent.com/62870568/220936710-b19f32eb-ecb6-43ff-b5de-939d8665d0dc.png)

Ob das zur Usability beiträgt, muss du wissen ;)
Wir werden es einmal so im belassen

**Comment by gis-eni** [comment]
PS: und zum Thema, Punkte: ich einmal prüfen, ob das grundsätzlich wieder machbar ist. Wie dringend ist diese Anfrage?

**Comment by k-placer** [comment]
Die Anfrage ist momentan nicht dringend. Ich kann mir vorstellen, dass die Kolleg*innen nach Einschulung auch mit dem Kreistool + Radius/Entfernung auskommen.

**Comment by gis-eni** [comment]
Mit der Implementierung wurde jetzt bereits begonnen.
